OlinkAnalyze/R/detect_data_type.R

```diff
@@ -0,0 +1,31 @@
+#' Help function checking whether the input data contains
+#' a column named "NPX" or "Quantified_value".
+#'
+#' @author
+#'  Simon Forsberg
+#'  Masoumeh Sheikhi
+#' @param df An arrow object
+#' @return A character string indicating the detected data type
+#' ("NPX" or "Quantified_value").
+#'
+detect_data_type <- function(df) {
+  # check if it's an arrow object, stop if not
+  check_is_arrow_object(df = df, error = TRUE)
+
+  # Extract column names
+  df_colnames <- names(df)
+
+  # Check whether df contains NPX or QUANT ----
+  if ("NPX" %in% df_colnames) {
```

Comment:
Also check that there is exactly 1 column name contaning NPX and only one column containing Quantified_value.